src/redux/actions/postActions.js

```diff
@@ -103,6 +108,55 @@ export const editPost = (slug, postData) => dispatch => {
       })
     );
 };
+export const reportArticleActionCreator = (res) => ({
+  type: REPORT_POST,
+  payload: res.data
+});
+export const reportArticle = (slug, postData) =>async dispatch => {
+  dispatch(clearErrors());
+  try {
+    const res = await reportArticleApi(slug, postData);
+    dispatch(reportArticleActionCreator(res));
+    toast.success('You have succesfully reported this article');
+  } catch(err){
+    toast.error(err.response.data.error);
+    dispatch({
+      type: REPORT_POST,
+      payload:null
+    });
+  }
+};
+
+export const getReports = () => dispatch => {
+  dispatch(setPostLoading());
+  return getreportsApi()
+    .then(res =>
+      dispatch({
+        type: GET_REPORTS,
+        payload: res.data
```

Comment:
Catch network errors here for all the actions. When there is no network, user should receive a message.

Reply:
@IssaIan let me see how I can do that

Reply:
@IssaIan fixed